### ISSUE TITLE
🐛 Bug Report: Scaffolder Actions 403 getting GitHub Commit status

### ISSUE THREAD

**Issue by awanlin** [backstage/backstage#17465]
### 📜 Description

After upgrading to `1.13.0` some of the Scaffolder actions are failing with a 403 error getting GitHub Commit status. So far the known actions with this error are - `fetch:plain:file` and `publish:github:pull-request`

**Note:** the solution if you are using a GitHub App is to give it at least read-only access to Commit statuses

![Screenshot 2023-04-19 at 8 42 03 AM](https://user-images.githubusercontent.com/67169551/233368023-9871901b-0455-446a-b90f-6b3fe033f6ce.png)

Related permission documentation: https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#statuses

### 👍 Expected behavior

Scaffolder actions should not throw a 403 error

### 👎 Actual Behavior with Screenshots

When running the certain Scaffolder actions you get this error:

```text
12023-04-19T04:16:54.466Z Beginning step Generating the Source Code Component22023-04-19T04:16:54.505Z info: Fetching template content from remote URL32023-04-19T04:16:54.657Z Error: Request failed for https://api.github.com/repos/***/***/commits/main/status?per_page=0, 403 Forbidden4    at _GithubUrlReader.fetchResponse (/app/node_modules/@backstage/backend-common/dist/index.cjs.js:2381:13)5    at runMicrotasks (<anonymous>)6    at processTicksAndRejections (node:internal/process/task_queues:96:5)7    at async _GithubUrlReader.fetchJson (/app/node_modules/@backstage/backend-common/dist/index.cjs.js:2386:22)8    at async _GithubUrlReader.getRepoDetails (/app/node_modules/@backstage/backend-common/dist/index.cjs.js:2357:26)9    at async _GithubUrlReader.readTree (/app/node_modules/@backstage/backend-common/dist/index.cjs.js:2255:25)10    at async UrlReaderPredicateMux.readTree (/app/node_modules/@backstage/backend-common/dist/index.cjs.js:3130:16)11    at async fetchContents (/app/node_modules/@backstage/plugin-scaffolder-backend/dist/cjs/ScaffolderEntitiesProcessor-2f24bb9e.cjs.js:565:17)12    at async Object.handler (/app/node_modules/@backstage/plugin-scaffolder-backend/dist/cjs/ScaffolderEntitiesProcessor-2f24bb9e.cjs.js:1006:7)
```

### 👟 Reproduction steps

Run any Software Template that is using the `fetch:plain:file` or the `publish:github:pull-request` action with GitHub

### 📃 Provide the context for the Bug.

This was an unexpected issue that we needed to suddenly resolve after upgrading as it wasn't an optional change and not clear in the release notes or changelog that could impact things in this way.

### 🖥️ Your Environment

The issue is strictly in `1.13.0` but here is my `info` details:

```text
OS:   Darwin 22.4.0 - darwin/arm64
node: v18.13.0
yarn: 1.22.19
cli:  0.22.6 (installed)
backstage:  1.13.0

Dependencies:
  @backstage/app-defaults                                   1.3.0
  @backstage/backend-app-api                                0.4.2
  @backstage/backend-common                                 0.18.4
  @backstage/backend-dev-utils                              0.1.1
  @backstage/backend-plugin-api                             0.5.1
  @backstage/backend-tasks                                  0.5.1
  @backstage/backend-test-utils                             0.1.36
  @backstage/catalog-client                                 1.4.1
  @backstage/catalog-model                                  1.3.0
  @backstage/cli-common                                     0.1.12
  @backstage/cli-node                                       0.1.0
  @backstage/cli                                            0.22.6
  @backstage/config-loader                                  1.2.0
  @backstage/config                                         1.0.7
  @backstage/core-app-api                                   1.7.0
  @backstage/core-components                                0.13.0
  @backstage/core-plugin-api                                1.5.1
  @backstage/dev-utils                                      1.0.14
  @backstage/errors                                         1.1.5
  @backstage/eslint-plugin                                  0.1.3
  @backstage/integration-aws-node                           0.1.2
  @backstage/integration-react                              1.1.12
  @backstage/integration                                    1.4.4
  @backstage/plugin-adr-backend                             0.3.2
  @backstage/plugin-adr-common                              0.2.8
  @backstage/plugin-adr                                     0.5.0
  @backstage/plugin-analytics-module-ga                     0.1.28
  @backstage/plugin-api-docs                                0.9.2
  @backstage/plugin-app-backend                             0.3.44
  @backstage/plugin-auth-backend                            0.18.2
  @backstage/plugin-auth-node                               0.2.13
  @backstage/plugin-azure-devops-backend                    0.3.23
  @backstage/plugin-azure-devops-common                     0.3.0
  @backstage/plugin-azure-devops                            0.2.8
  @backstage/plugin-badges-backend                          0.1.38
  @backstage/plugin-badges                                  0.2.41
  @backstage/plugin-catalog-backend-module-bitbucket-server 0.1.9
  @backstage/plugin-catalog-backend-module-github           0.2.7
  @backstage/plugin-catalog-backend                         1.9.0
  @backstage/plugin-catalog-common                          1.0.13
  @backstage/plugin-catalog-graph                           0.2.29
  @backstage/plugin-catalog-import                          0.9.7
  @backstage/plugin-catalog-node                            1.3.5
  @backstage/plugin-catalog-react                           1.5.0
  @backstage/plugin-catalog                                 1.10.0
  @backstage/plugin-entity-feedback-backend                 0.1.2
  @backstage/plugin-entity-feedback-common                  0.1.1
  @backstage/plugin-entity-feedback                         0.2.0
  @backstage/plugin-events-backend-module-azure             0.1.6
  @backstage/plugin-events-backend-module-bitbucket-cloud   0.1.6
  @backstage/plugin-events-backend-module-github            0.1.6
  @backstage/plugin-events-backend                          0.2.5
  @backstage/plugin-events-node                             0.2.5
  @backstage/plugin-explore-common                          0.0.1
  @backstage/plugin-explore-react                           0.0.28
  @backstage/plugin-explore                                 0.4.2
  @backstage/plugin-github-actions                          0.5.17
  @backstage/plugin-github-pull-requests-board              0.1.11
  @backstage/plugin-home                                    0.5.0
  @backstage/plugin-kubernetes-backend                      0.10.0
  @backstage/plugin-kubernetes-common                       0.6.2
  @backstage/plugin-kubernetes                              0.8.0
  @backstage/plugin-org                                     0.6.7
  @backstage/plugin-permission-backend                      0.5.19
  @backstage/plugin-permission-common                       0.7.5
  @backstage/plugin-permission-node                         0.7.7
  @backstage/plugin-permission-react                        0.4.12
  @backstage/plugin-playlist-backend                        0.3.0
  @backstage/plugin-playlist-common                         0.1.6
  @backstage/plugin-playlist                                0.1.8
  @backstage/plugin-proxy-backend                           0.2.38
  @backstage/plugin-scaffolder-backend                      1.13.0
  @backstage/plugin-scaffolder-common                       1.2.7
  @backstage/plugin-scaffolder-node                         0.1.2
  @backstage/plugin-scaffolder-react                        1.3.0
  @backstage/plugin-scaffolder                              1.13.0
  @backstage/plugin-search-backend-module-catalog           0.1.0
  @backstage/plugin-search-backend-module-elasticsearch     1.2.0
  @backstage/plugin-search-backend-module-pg                0.5.5
  @backstage/plugin-search-backend-module-techdocs          0.1.0
  @backstage/plugin-search-backend-node                     1.2.0
  @backstage/plugin-search-backend                          1.3.0
  @backstage/plugin-search-common                           1.2.3
  @backstage/plugin-search-react                            1.5.2
  @backstage/plugin-search                                  1.2.0
  @backstage/plugin-shortcuts                               0.3.9
  @backstage/plugin-sonarqube-backend                       0.1.9
  @backstage/plugin-sonarqube-react                         0.1.5
  @backstage/plugin-sonarqube                               0.6.6
  @backstage/plugin-tech-insights-backend-module-jsonfc     0.1.28
  @backstage/plugin-tech-insights-backend                   0.5.10
  @backstage/plugin-tech-insights-common                    0.2.10
  @backstage/plugin-tech-insights-node                      0.4.2
  @backstage/plugin-tech-insights                           0.3.9
  @backstage/plugin-tech-radar                              0.6.3
  @backstage/plugin-techdocs-backend                        1.6.1
  @backstage/plugin-techdocs-module-addons-contrib          1.0.12
  @backstage/plugin-techdocs-node                           1.7.0
  @backstage/plugin-techdocs-react                          1.1.5
  @backstage/plugin-techdocs                                1.6.1
  @backstage/plugin-user-settings                           0.7.2
  @backstage/release-manifests                              0.0.9
  @backstage/test-utils                                     1.3.0
  @backstage/theme                                          0.2.19
  @backstage/types                                          1.0.2
  @backstage/version-bridge                                 1.0.4
✨  Done in 1.87s.
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by awanlin** [comment]
FYI @fabidick22 (on Discord: https://discord.com/channels/687207715902193673/1098110316149223575) and @ArtemChekunov (here on GitHub: https://github.com/backstage/backstage/issues/17385#issuecomment-1514683774) also had this issue

**Comment by awanlin** [comment]
Oh, and I think maybe this PR is the change that introduced this: https://github.com/backstage/backstage/pull/17233

**Comment by agarthetiger** [comment]
We're seeing a very similar (related?) issue when trying to access tech docs with v1.13. On the docs page there is an error saying 
`Building a newer version of this documentation failed. Error: "Request failed for https://api.github.com/repos/<org-redacted>/<repo-redacted>/commits/main/status?per_page=0, 403 Forbidden"`

**Comment by awanlin** [comment]
Bummer @agarthetiger, have you tried the fix mentioned? Adding the read-only permission for Commit statuses to your GitHub App?

**Comment by agarthetiger** [comment]
I selected all three options when I ran `yarn backstage-cli create-github-app <org>`, Read access to content, Read access to members, and Read and Write to content and actions. In GitHub under the app settings I see the Permissions for `Read access to members and metadata`, `Read and write access to actions and code`, and repository access for All repositories selected.

Is there something else you mean? I've read this again, https://backstage.io/docs/integrations/github/github-apps#app-permissions but I don't see anything else I can configure?

**Comment by freben** [comment]
That last one (with `?per_page=0`) originates from #17298, ping @pjungermann

**Comment by awanlin** [comment]
@agarthetiger it's not in the docs yet as this is something new that just happened since the `1.13.0` release a few days ago. So in the GitHub App configuration area for permission on GitHub you need to update it to be like this:

![Screenshot 2023-04-19 at 8 42 03 AM](https://user-images.githubusercontent.com/67169551/233368023-9871901b-0455-446a-b90f-6b3fe033f6ce.png)

**Comment by agarthetiger** [comment]
Thank you @awanlin , giving Read-only access to the commit Statuses has fixed it for me.

**Comment by tibbing** [comment]
For anyone else struggling with this, don't forget to have the account owner approve the new permissions after giving it read-only access to the commit statuses (an email should be sent out).

**Comment by alanhughes** [comment]
Just in case anyone else is googling this, we had the same issue just now, but it manifested via techdocs, rather than scaffolder. The error we saw was:
```
[1] 2023-04-27T11:05:01.825Z techdocs error Failed to build the docs page: Request failed for https://api.github.com/repos/<org>/<repo>/commits/master/status?per_page=0, 403 Forbidden Request failed for https://api.github.com/repos/<org>/<repo>/commits/master/status?per_page=0, 403 Forbidden type=plugin stack=Error: Request failed for https://api.github.com/repos/<org>/<repo>/commits/master/status?per_page=0, 403 Forbidden
```
Updating the app permissions as detailed above resolved it.

**Comment by awanlin** [comment]
Closing as resolved by #17472
